### PR TITLE
Move imagenet inference to nightly

### PR DESCRIPTION
**Pull request by ChaiBapchya**
## Description ##
Reduces the timeout issue for UNIX GPU by moving the imagenet inference

Pointed out by @aaronmarkham 
Imagenet inference test (along with few others) involve downloading 1.5G of 

### Essentials ###

- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- [x] Code is well-documented: 
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Remove test from unix GPU cpp examples
- [x] Add test to nightly

## Comments ##
- Tested it on MXNet CI Dev Jenkins
http://jenkins.mxnet-ci-dev.amazon-ml.com/blue/organizations/jenkins/NightlyTestsForBinaries/detail/move_imagenet_inference_nightly/18/pipeline

### CODE REVIEW

File: ci/docker/runtime_functions.sh [hunk 1]
```diff
@@ -1391,6 +1391,15 @@ nightly_test_installation() {
     source ./tests/jenkins/run_test_installation_docs.sh docs/install/index.md 1 1686; ${1}
 }
 
+# Runs Imagenet inference
+nightly_test_imagenet_inference() {
+    set -ex
+    echo $PWD
+    cp /work/mxnet/build/cpp-package/example/imagenet_inference .
```
Comment (marcoabreu) [inline ci/docker/runtime_functions.sh]: Is this tested?

Comment (ChaiBapchya) [reply ci/docker/runtime_functions.sh]: yes. on ci dev. it exists.

**Comment by ChaiBapchya** [comment]
Fixes #16524

**Comment by ChaiBapchya** [comment]
@aaronmarkham Plz take a look. Thanks.

File: ci/docker/runtime_functions.sh (hunk 1 above)
Comment (ChaiBapchya) [reply ci/docker/runtime_functions.sh]: Spoke prematurely.
http://jenkins.mxnet-ci-dev.amazon-ml.com/blue/organizations/jenkins/NightlyTestsForBinaries/detail/move_imagenet_inference_nightly/20/pipeline/72

While it passed the `cp ...` line, 

Failed at this line in 
`./imagenet_inference --symbol_file ./model/Inception-BN-symbol.json --params_file ./model/Inception-BN-0126.params --dataset ./data/val_256_q90.rec --rgb_mean '123.68 116.779 103.939' --batch_size 1 --num_skipped_batches 50 --num_inference_batches 500`

Error being
```
./imagenet_inference: error while loading shared libraries: libmxnet.so: cannot open shared object file: No such file or directory
```

Unable to understand despite me adding 

https://github.com/ChaiBapchya/incubator-mxnet/blob/daa88e6ff8f4e98742ea46bbe77085147bf2aeca/tests/nightly/JenkinsfileForBinaries#L23

**Pull request merged**